app/controllers/microcosms_controller.rb

```diff
@@ -0,0 +1,61 @@
+class MicrocosmsController < ApplicationController
+  layout "site"
+  before_action :authorize_web
+
+  before_action :set_microcosm, :only => [:edit, :show, :show_events, :show_members, :update]
+
+  authorize_resource
+
+
+  def index
+    @microcosms = Microcosm.order(:name)
+  end
+
+  # GET /microcosms/mycity
+  # GET /microcosms/mycity.json
+  def show
+  end
+
+  def show_members
+  end
+
+  def show_events
+  end
+
+  def edit
+  end
+
+  def update
+    respond_to do |format|
+      if @microcosm.update(microcosm_params)
+        format.html { redirect_to @microcosm, notice: 'Microcosm was successfully updated.' }
```

Comment:
probably lots of those strings would be better off in locales for i18n.

Reply:
Got the translation in there.